site/js/modules/webui/course.js

```diff
@@ -35,12 +39,140 @@ function handlerCourse(path, params, context, container) {
+                        </div>
+                        <div class="checkbox-field">
+                            <input type="checkbox" id="dry-run" name="dry-run" />
+                            <label for="dry-run">Send as Dry Run</label>
+                        </div>
+                        <div class="checkbox-field">
+                            <input type="checkbox" id="html-format" name="html-format" />
+                            <label for="html-format">Content is in HTML</label>
+                        </div>
+                    </fieldset>
+                </div>
+                <button class="send-email">Send Email</button>
+                <div class="results-area"></div>
+            </div>
+        </div>
+    `;
+
+    document.querySelector('button.send-email').addEventListener('click', function(event) {
+        Routing.loadingStart(container.querySelector(".results-area"), false);
+
+        let args = {
+            [Routing.PARAM_COURSE_ID]: course.id,
+            [Routing.PARAM_EMAIL_TO]: extractRecipients(container.querySelector("fieldset [name='to']").value),
+            [Routing.PARAM_EMAIL_CC]: extractRecipients(container.querySelector("fieldset [name='cc']").value),
+            [Routing.PARAM_EMAIL_BCC]: extractRecipients(container.querySelector("fieldset [name='bcc']").value),
+            [Routing.PARAM_EMAIL_SUBJECT]: container.querySelector("fieldset [name='subject']").value,
+            [Routing.PARAM_EMAIL_BODY]: container.querySelector("fieldset [name='body']").value,
+            [Routing.PARAM_DRY_RUN]: container.querySelector("fieldset [name='dry-run']").checked,
+            [Routing.PARAM_EMAIL_HTML]: container.querySelector("fieldset [name='html-format']").checked,
+        };
+
+        if (args[Routing.PARAM_EMAIL_TO].length === 0 &&
+                args[Routing.PARAM_EMAIL_CC].length === 0 &&
+                args[Routing.PARAM_EMAIL_BCC].length === 0) {
+            renderResult('<p>Please specify at least one recipient.</p>');
+            return;
+        }
+
+        if (args[Routing.PARAM_EMAIL_SUBJECT].length === 0) {
+            renderResult('<p>Please include a subject.</p>');
+            return;
+        }
+
+        Autograder.Course.email(args)
+            .then(function(result) {
+                renderResult(`
+                    <p>Email sent successfully.</p>
+                    <p><strong>Recipients</strong> (total ${result[Routing.PARAM_EMAIL_TO].length})</p>
+                    <p>${result[Routing.PARAM_EMAIL_TO].join(', ')}</p>
+                    <p><strong>CC-ed Recipients</strong> (total ${result[Routing.PARAM_EMAIL_CC].length})</p>
+                    <p>${result[Routing.PARAM_EMAIL_CC].join(', ')}</p>
+                    <p><strong>BCC-ed Recipients</strong> (total ${result[Routing.PARAM_EMAIL_BCC].length})</p>
+                    <p>${result[Routing.PARAM_EMAIL_BCC].join(', ')}</p>
```

Comment:
This part seems a bit rough.
In cases with a few recipients, it seems overkill.
In cases with many recipients, it seems squashed (since all recipient type will be on one line).

What about just using the raw JSON?
I'm definitely biased, but that already has things organized.